### DISCUSSION TITLE
I can't build the TTGO-T-Display board.

### DISCUSSION THREAD

**Discussion by RedSalsicha** [BitMaker-hub/NerdMiner_v2#328] [Q&A]
I can't build with the TTGO-T-Display board.

PlatformIO Build generates these errors that appear to be missing the "sdkconfig.h" file in the Platformio folders, but when searching online for alternative solutions, what I found didn't work.

> ` *  Executing task: C:\Users\Win10\.platformio\penv\Scripts\platformio.exe run --environment TTGO-T-Display 

Processing TTGO-T-Display (platform: espressif32; board: esp32dev; framework: arduino)
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (6.5.0) > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20014.231204 (2.0.14)
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 42 compatible libraries
Scanning dependencies...
Dependency Graph
|-- OpenFontRender @ 1.1.0+sha.f163cc6
|-- ArduinoJson @ 6.21.5
|-- WiFiManager @ 2.0.16-rc.2+sha.88c378f
|-- OneButton @ 2.5.0
|-- NTPClient @ 3.2.1
|-- WebServer @ 2.0.0
|-- WiFi @ 2.0.0
|-- rm67162
|-- TFT_eSPI @ 2.5.23
|-- SPI @ 2.0.0
|-- FS @ 2.0.0
|-- SD @ 2.0.0
|-- SD_MMC @ 2.0.0
|-- SPIFFS @ 2.0.0
|-- HTTPClient @ 2.0.0
Building in release mode
Compiling .pio\build\TTGO-T-Display\src\NerdMinerV2.ino.cpp.o
Compiling .pio\build\TTGO-T-Display\src\ShaTests\nerdSHA256.cpp.o
Compiling .pio\build\TTGO-T-Display\src\ShaTests\nerdSHA256plus.cpp.o
In file included from c:\users\win10\.platformio\packages\toolchain-xtensa-esp32\xtensa-esp32-elf\sys-include\sys\reent.h:503,
                 from C:/Users/Win10/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from c:\users\win10\.platformio\packages\toolchain-xtensa-esp32\xtensa-esp32-elf\sys-include\stdio.h:60,
Compiling .pio\build\TTGO-T-Display\src\drivers\displays\amoledDisplayDriver.cpp.o
                 from src/ShaTests/nerdSHA256plus.cpp:16:
C:/Users/Win10/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/newlib/platform_include/assert.h:20:10: fatal error: sdkconfig.h: No such file or directory

*******************************************************************
* Looking for sdkconfig.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:sdkconfig.h"
* Web  > https://registry.platformio.org/search?q=header:sdkconfig.h
*
*******************************************************************

 #include <sdkconfig.h>
          ^~~~~~~~~~~~~
compilation terminated.
In file included from c:\users\win10\.platformio\packages\toolchain-xtensa-esp32\xtensa-esp32-elf\sys-include\sys\reent.h:503,
                 from C:/Users/Win10/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from c:\users\win10\.platformio\packages\toolchain-xtensa-esp32\xtensa-esp32-elf\sys-include\stdio.h:60,
                 from src/ShaTests/nerdSHA256.cpp:2:
C:/Users/Win10/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/newlib/platform_include/assert.h:20:10: fatal error: sdkconfig.h: No such file or directory  

*******************************************************************
* Looking for sdkconfig.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:sdkconfig.h"
* Web  > https://registry.platformio.org/search?q=header:sdkconfig.h
*
*******************************************************************

 #include <sdkconfig.h>
          ^~~~~~~~~~~~~
compilation terminated.
In file included from c:\users\win10\.platformio\packages\toolchain-xtensa-esp32\xtensa-esp32-elf\sys-include\sys\reent.h:503,
                 from C:/Users/Win10/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from c:\users\win10\.platformio\packages\toolchain-xtensa-esp32\xtensa-esp32-elf\sys-include\stdio.h:60,
                 from C:/Users/Win10/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:27,
                 from src/NerdMinerV2.ino.cpp:3:
C:/Users/Win10/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/newlib/platform_include/assert.h:20:10: fatal error: sdkconfig.h: No such file or directory  

*******************************************************************
* Looking for sdkconfig.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:sdkconfig.h"
* Web  > https://registry.platformio.org/search?q=header:sdkconfig.h
*
*******************************************************************

 #include <sdkconfig.h>
          ^~~~~~~~~~~~~
compilation terminated.
*** [.pio\build\TTGO-T-Display\src\ShaTests\nerdSHA256plus.cpp.o] Error 1
*** [.pio\build\TTGO-T-Display\src\ShaTests\nerdSHA256.cpp.o] Error 1
*** [.pio\build\TTGO-T-Display\src\NerdMinerV2.ino.cpp.o] Error 1
============================================================================ [FAILED] Took 15.50 seconds ============================================================================

Environment     Status    Duration
--------------  --------  ------------
TTGO-T-Display  FAILED    00:00:15.496
======================================================================= 1 failed, 0 succeeded in 00:00:15.496 ======================================================================= 

 *  The terminal process "C:\Users\Win10\.platformio\penv\Scripts\platformio.exe 'run', '--environment', 'TTGO-T-Display'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 
`

**Comment by sany3001** [comment]
I have reported similar issue: https://github.com/BitMaker-hub/NerdMiner_v2/issues/299

**Comment by sany3001** [comment]
I have tried to search for that file on GitHub and it is even more strange: only one location, really?

https://github.com/search?q=repo%3ABitMaker-hub%2FNerdMiner_v2+sdkconfig.h&type=code

**Reply by jsy-rg1570** [reply]
Dear,

I answered here : https://github.com/BitMaker-hub/NerdMiner_v2/issues/299#issuecomment-1994383621

**Reply by sany3001** [reply]
Thank you!!
